### PR TITLE
feat: Full text search

### PR DESCRIPTION
**Pull request by mugikhan**
## Work Done
Add full text search to react supabase todo list example
Add recursive triggers for FTS
Add performance tests for wa-sqlite

## How to test
Follow readme on how to get react app up and running. FTS will be supported out of the box.

## Screenshots
Screenshot 2024-04-17 at 10 26 03
Screenshot 2024-04-17 at 10 26 12

### CODE REVIEW

File: demos/react-supabase-todolist/src/components/providers/SystemProvider.tsx [hunk 1]
```diff
@@ -6,6 +6,8 @@ import { WASQLitePowerSyncDatabaseOpenFactory } from '@journeyapps/powersync-sdk
 import { CircularProgress } from '@mui/material';
 import Logger from 'js-logger';
 import React, { Suspense } from 'react';
+// For console testing purposes
```
Comment (stevensJourney) [inline demos/react-supabase-todolist/src/components/providers/SystemProvider.tsx:9]: This comment seems to have moved.

File: demos/react-supabase-todolist/src/components/widgets/SearchBarWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  const [searchResults, setSearchResults] = React.useState<SearchResult[]>([]);
+  const [value, setValue] = React.useState<SearchResult | null>(null);
+
+  const navigate = useNavigate();
+
+  const handleInputChange = async (value: string) => {
+    if (value.length !== 0) {
+      let listsSearchResults = await searchTable(value, 'lists');
+      let todoItemsSearchResults = await searchTable(value, 'todos');
+      let formattedListResults: SearchResult[] = listsSearchResults.map(
```
Comment (stevensJourney) [inline demos/react-supabase-todolist/src/components/widgets/SearchBarWidget.tsx:18]: It would be nice to distinguish between todos and lists here.

Comment (mugikhan) [reply demos/react-supabase-todolist/src/components/widgets/SearchBarWidget.tsx:31]: Updated it to include the list name with todos

File: demos/react-supabase-todolist/src/components/providers/SystemProvider.tsx (hunk 1 above)
Comment (mugikhan) [reply demos/react-supabase-todolist/src/components/providers/SystemProvider.tsx]: Reverted it back.

File: demos/react-supabase-todolist/src/components/widgets/SearchBarWidget.tsx (hunk 2 above)
Comment (stevensJourney) [reply demos/react-supabase-todolist/src/components/widgets/SearchBarWidget.tsx:31]: This looks nice!

**Review by stevensJourney — DISMISSED** [review]
Looks good to me. Remember to bump the WA-SQLite packages before merging

File: demos/react-supabase-todolist/src/app/utils/fts_setup.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import { AppSchema } from '../../library/powersync/AppSchema';
+import { Table } from '@journeyapps/powersync-sdk-web';
+import { db } from '@/components/providers/SystemProvider';
+import { ExtractType, generateJsonExtracts } from './helpers';
+
+/// Create a Full Text Search table for the given table and columns
```
Comment (benitav) [inline demos/react-supabase-todolist/src/app/utils/fts_setup.ts:6]: Is this the correct syntax for comments? Shouldn't it be JSDoc?

Comment (mugikhan) [reply demos/react-supabase-todolist/src/app/utils/fts_setup.ts]: Updated the comments.

**Review by stevensJourney — APPROVED** [review]
Nice! I love the demo app's search function. Looks amazing.

**Pull request merged**